### ISSUE TITLE
Why is the inference speed of Yolov5 PyTorch model (with raw uint8 images) better than ONNX and OpenVino models

### ISSUE THREAD

**Issue by arpane4c5** [ultralytics/yolov5#6607]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to improve the inference time of yolov5s model using OpenVino for a CPU based Intel i7 deployment. The method I am following is same as given [here](https://docs.openvino.ai/latest/notebooks/102-pytorch-onnx-to-openvino-with-output.html). I have a trained model of yolov5s, which was custom trained on our dataset using the `train.py` script. Then I exported the model into ONNX and OpenVino in the following way (Note that the image size is (500, 2048, 3) (H, W, C). 

Loading the PyTorch model
```
# Create pytorch yolov5s model
pytorch_model = torch.hub.load(model_path_prefix, 
                               "custom", 
                               path= pytorch_model_path, 
                               source = 'local', 
                               device='cpu')
```
Output:
```
YOLOv5 🚀 v6.0-238-gb57abb1 torch 1.10.1+cu113 CPU

Fusing layers... 
Model Summary: 213 layers, 7012822 parameters, 0 gradients, 15.8 GFLOPs
Adding AutoShape... 
```
Exporting to ONNX (Tried two ways) 
```
if not os.path.exists(onnx_model_path):
    dummy_input = torch.randn(1, 3, 500, 2048)
    torch.onnx.export(
        pytorch_model,
        dummy_input,
        onnx_model_path,
        opset_version=11,
    )
    print(f"ONNX model exported to {onnx_model_path}.")
else:
    print(f"ONNX model {onnx_model_path} already exists.")
```

`python export.py --weights yolov5s_our_custom_model.pt --include onnx --img 500 2048 --int8`

Performance Comparison on 20 images
```
num_images = 20

start = time.perf_counter()
for _ in range(num_images):
    exec_net_onnx.infer(inputs={input_layer_onnx: np.expand_dims(np.transpose(resized_image, (2, 0, 1)), 0)})
end = time.perf_counter()
time_onnx = end - start
print(
    f"ONNX model in Inference Engine/CPU: {time_onnx/num_images:.3f} "
    f"seconds per image, FPS: {num_images/time_onnx:.2f}"
)

start = time.perf_counter()
for _ in range(num_images):
    exec_net_ir.infer(inputs={input_layer_ir: normalized_input_image})
end = time.perf_counter()
time_ir = end - start
print(
    f"IR model in Inference Engine/CPU: {time_ir/num_images:.3f} "
    f"seconds per image, FPS: {num_images/time_ir:.2f}"
)

with torch.no_grad():
    start = time.perf_counter()
    for _ in range(num_images):
        #pytorch_model(torch.as_tensor(normalized_input_image).float())
        pytorch_model(img)
    end = time.perf_counter()
    time_torch = end - start
print(
    f"PyTorch model on CPU (passing raw images): {time_torch/num_images:.3f} seconds per image, "
    f"FPS: {num_images/time_torch:.2f}"
)

with torch.no_grad():
    start = time.perf_counter()
    for _ in range(num_images):
        pytorch_model(torch.as_tensor(normalized_input_image).float())
        #pytorch_model(img)
    end = time.perf_counter()
    time_torch = end - start
print(
    f"PyTorch model on CPU (passing torch Tensor images): {time_torch/num_images:.3f} seconds per image, "
    f"FPS: {num_images/time_torch:.2f}"
)
```
Output:
```
ONNX model in Inference Engine/CPU: 0.109 seconds per image, FPS: 9.18
IR model in Inference Engine/CPU: 0.106 seconds per image, FPS: 9.46
PyTorch model on CPU (passing raw images): 0.040 seconds per image, FPS: 24.91
PyTorch model on CPU (passing torch Tensor images): 0.373 seconds per image, FPS: 2.68
```

I have also tried the following ways of converting the models into ONNX and OpenVino but the result does not change much. 
```
root@54c9a540e010:/workspace/VisionWorkspace/yolov5# python export.py --weights /workspace/VisionWorkspace/yolov5s_our_custom_model.pt --imgsz 500 2048 --include openvino
export: data=data/coco128.yaml, weights=['/workspace/VisionWorkspace/yolov5s_our_custom_model.pt'], imgsz=[500, 2048], batch_size=1, device=cpu, half=False, inplace=False, train=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['openvino']
YOLOv5 🚀 v6.0-238-gb57abb1 torch 1.10.1+cu113 CPU

Fusing layers... 
Model Summary: 213 layers, 7012822 parameters, 0 gradients, 15.8 GFLOPs
WARNING: --img-size 500 must be multiple of max stride 32, updating to 512

PyTorch: starting from /workspace/VisionWorkspace/yolov5s_our_custom_model.pt with output shape (1, 64512, 6) (14.3 MB)

ONNX: starting export with onnx 1.8.204...
ONNX: export success, saved as /workspace/VisionWorkspace/yolov5s_our_custom_model.onnx (29.1 MB)

OpenVINO: starting export with openvino 2021.4.2-3976-0943ed67223-refs/pull/539/head...
OpenVINO: export success, saved as /workspace/VisionWorkspace/yolov5s_our_custom_model_openvino_model/ (29.3 MB)

Export complete (12.61s)
Results saved to /workspace/VisionWorkspace
Detect:          python detect.py --weights /workspace/VisionWorkspace/yolov5s_our_custom_model_openvino_model/
PyTorch Hub:     model = torch.hub.load('ultralytics/yolov5', 'custom', '/workspace/VisionWorkspace/yolov5s_our_custom_openvino_model/')
Validate:        python val.py --weights /workspace/VisionWorkspace/yolov5s_our_custom_openvino_model/
Visualize:       https://netron.app

```

My question is that, when I convert my input image to a `torch.as_tensor(normalized_input_image).float()` and then pass to the PyTorch model, the performance drops, but if I feed the `uint8` image as it is to the PyTorch model the performance improves a lot. Is there an underlying difference during inference on a float() image as compared to the uint8 image. Is the `AutoShape' that is given added by the PyTorch model having an effect in performance. What optimizations are needed for the OpenVino and ONNX model to achieve such an improved performance. 

Thanks,
Arpan

**Comment by glenn-jocher** [comment]
@arpane4c5 **results will vary widely by software and hardware**, but in general CPUs **may** show slight improvements in inference time for ONNX, OpenVINO vs PyTorch. You can use the new benchmarks branch for this:

https://github.com/ultralytics/yolov5/blob/updates/benchmarks/utils/benchmarks.py

```
benchmarks: weights=/usr/src/yolov5/yolov5s.pt, imgsz=640, batch_size=1, data=/usr/src/yolov5/data/coco128.yaml
...

Benchmarks complete (782.19s)
                   Weights  mAP@0.5:0.95  Inference time (ms)
0               yolov5s.pt      0.407554           102.777191
1      yolov5s.torchscript      0.402908           132.848348
2             yolov5s.onnx      0.402908            89.061137
3   yolov5s_openvino_model      0.402908            67.093970
4                   engine           NaN                  NaN
5                   coreml           NaN                  NaN
6      yolov5s_saved_model      0.402908           133.983964
7               yolov5s.pb      0.402908           101.405423
8      yolov5s-fp16.tflite      0.402851           502.289245
9                  edgetpu           NaN                  NaN
10                    tfjs           NaN                  NaN
```

**Comment by arpane4c5** [comment]
Thank you @glenn-jocher for your reply. I was also expecting slight improvements in inference time for ONNX, OpenVino over PyTorch and the same is being achieved when I send the input image after being converted to a float tensor, to the 3 models (9.18, 9.46 and 2.68 FPS, respectively). But these performances are worse when I directly use a uint8 image as input to the PyTorch model (24.91 FPS). I cannot understand the reason why this is the case.
